test/integration/navigation_test.rb

```diff
@@ -3,7 +3,57 @@
 require 'test_helper'
 
 class NavigationTest < ActionDispatch::IntegrationTest
-  # test "the truth" do
-  #   assert true
-  # end
+  fixtures :users
+
+  test 'failed access, sign in and redirect to protected resource' do
+    alice = users(:alice)
+
+    # Verify the user has no access to the /secret endpoint and
+    # is instead redirected to homepage. Meanwhile access to /secret is
+    # stored in the users session.
+    get '/secret'
+    assert_equal 302, status
+    assert_equal 'Not worthy!', flash['error']
+    follow_redirect!
+    assert_equal 200, status
+    assert_equal '/', path
+
+    # Load login form
+    get '/users/sign_in'
+    assert_equal 200, status
+
+    # Submit form
+    post '/users/sign_in',
+      params: {
+        passwordless: { email: alice.email }
+      },
+      headers: { 'HTTP_USER_AGENT' => 'Mosaic v.1' }
+    assert_equal 200, status
+    assert response.body.include?(
+      'If we found you in the system, we have sent you an email.'
```

Comment:
`response.body.include?('If we found you in the system')` could be another solution to the `'` business.

Reply:
Yea.. I tried swapping the i18n backend/translations or stubbing the calls to it but everything turned out to be extremely cumbersome for this single case.

So eventually I like to go w/ your proposed way of checking for the string partial.